crates/bindings-csharp/BSATN.Runtime/Builtins.cs

```diff
@@ -61,67 +70,69 @@ string wrapperPropertyName
         );
 }
 
-public record Address : BytesWrapper
+public readonly record struct Address
 {
-    protected override int SIZE => 16;
+    private readonly U128 value;
 
     public Address() { }
-
-    private Address(byte[] bytes)
-        : base(bytes) { }
-
-    public static Address? From(byte[] bytes)
-    {
-        if (bytes.All(b => b == 0))
+    internal Address(U128 v) => value = v;
+    public static Address? From(ulong upper, ulong lower) {
+        if (upper == 0 && lower == 0)
         {
             return null;
         }
-        return new(bytes);
+        else
+        {
+            return new Address(new U128(upper, lower));
+        }
     }
-
-    public static Address Random()
-    {
-        var random = new Random();
-        var addr = new Address();
-        random.NextBytes(addr.bytes);
-        return addr;
+    public static Address? From(byte[] bytes) {
+        Debug.Assert(bytes.Length == 16);
+        return new Address(MemoryMarshal.Read<U128>(bytes));
```

Comment:
This still needs to check that given bytes do not represent a zero address, like before.

You should be able to do this via simply

```cs
var addr = new Address(...);
return addr == default ? null : addr;
```